URIs.md

```diff
@@ -0,0 +1,103 @@
+
+Following the chain ID may be a [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) account specifier, a [CAIP-19](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md) asset specifier, a transaction identifier, or other unspecified identifiers for on-chain resources and activity.
+
+```
+chain://<network>
+chain://<network>:<address>
+chain://<network>/<asset_namespace>:<asset_reference>/...
+chain://<network>/tx:<tx_identifier>
+chain://<network>/<future_placeholder>
+
+network             = CAIP-2 chain id
+address             = an address string in the common format of the given network
+asset_namespace     = an identifier for a class of related assets, such as erc20 and slip44
+asset_reference     = identifies an asset within its asset_namespace
+tx_identifier       = a transaction identifier in the common format of the given network
+future_placeholder  = future message types
+...                 = addition uri parameters for the standard (optional)
+```
+
+Examples:
+
+```
+chain://eip155:1:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03
+chain://eip155:1/erc721:0xaba7161a7fb69c88e16ed9f455ce62b791ee4d03/7894
+chain://bip122:000000000019d6689c085ae165831e93/slip44:0
+```
+
+### Farcaster Resources
+
+Farcaster URIs must be capable of identifying any message signed by a user. A sample URI might look like `farcaster://user:alice/cast:0xf00b4r/42` and is of the form:
+
+```
+farcaster://<user>
+farcaster://<user>/<message_type>:<message_id>
+farcaster://<user>/<message_type>:<message_id>/<sequence>
+
+user                = "user":<username> | "address":<address>
+username            = "[a-zA-Z0-9-]{1,16}"
+address             = "0x[a-fA-F0-9]{40}"
+message_type        = "cast" | "reaction" | "verified_address" | <future_placeholder>
+message_id          = unique identifier for the message
+sequence            = number (optional)
+future_placeholder  = future message types
+```
+
+The `sequence` property must be set when referring to a type that has a sequence, like a Cast or Reaction.
+
+Example:
+
+```
+farcaster://user:alice/cast:0xf00b4r/42
+```
+
+## Versioning
+
+If an existing resource or activity type already represented by Farcaster URIs must be updated for any reason, the same URI structure must continue to be supported alongside its upgraded version. A hyphenated suffix should describe the version upgrade. For example:
+
+```
+farcaster://user:alice/cast-v2:0xf00b4r/42/1652945621
+```
+
+might represent an update to the URI schema Farcaster Casts that adds a unix epoch timestamp to follow `<sequence>` in the original schema.
+
+## Extending
+
+The Farcaster URI standard may be extended by the community via updates to this document. The hierarchical nature of URIs allows for the addition of new identifier types at any point in the schema.
```

Comment:
it would be useful to have some formal definition that must be updated whenever a change is made. for example, a typescript method that will evaluate any input string and declare if it is valid or invalid.

Reply:
Agreed. I think that a good way to capture this requirement might be to state that updates to this doc must also come with a corresponding update to the URL parser in the reference Farcaster node implementation.